src/merkle/smt/mod.rs

```diff
@@ -446,6 +520,149 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
         value: &Self::Value,
     ) -> Self::Leaf;
 
+    /// Computes leaves from a set of key-value pairs and current leaf values.
+    /// Deried from `sorted_pairs_to_leaves`
+    ///
+    /// TODO: refactor and merge functionality with `sorted_pairs_to_leaves`?
```

Comment:
Yeah I think if we can merge this with `sorted_pairs_to_leaves`, that'd be great for deduplication.